terraform/shared/scripts/users.py

```diff
@@ -0,0 +1,40 @@
+import elasticsearch
+import secrets
+
+"""
+This method checks if a user exists.
+Exists: it updates the roles if needed with no password update, returns None
+Missing: creates the user with a new password , returns username, password
+"""
+
+
+def put_user_safely(es, username, roles):
+    try:
+        data = es.security.get_user(username=username)
+        print(f"user: {username} exists")
+        if data[username]["roles"] != roles:
+            print(f"user: {username} updating roles...")
+            es.security.put_user(
+                username=username,
+                body={
+                    "roles": roles
+                }
+            )
+            print(f"user: {username} roles updated")
+        else:
+            print(f"user: {username} has no role updates")
+
+        return None
+
+    except elasticsearch.exceptions.NotFoundError:
```

Comment:
General good Python practice is to wrap `try … except` blocks as tight as possible – in this case, around the `data = es.security.get_user(…)` line, so if this is thrown unexpectedly elsewhere it still raises.